README.md

```diff
@@ -254,6 +254,36 @@ Some rebasebot arguments are available in lifecycle hook scripts as environment
 
 *Note: Remotes are always `source`, `dest`, and `rebase`. The local branch is called rebase.*
 
+## Dynamic source branch selection
+
+Rebasebot offers the capability to dynamically determine the source branch or tag for rebasing, enabling integration into CI/CD pipelines. This is especially valuable when rebasing from release tags or dynamically created branches.
+
+Instead of statically specifying the `--source` parameter, utilize `--source-repo` in conjunction with `--source-branch-hook`. The script provided to `--source-branch-hook` is executed before the rebasebot git workspace setup. Its standard output must be the name of the branch or tag to be used as the rebase source.
+
+You can implement any logic within the script to select the desired source branch, such as querying the GitHub API for the latest release or .
+
+Possible formats for path to the script:
+* Local filepath
+* Builtin with `_BUILTIN_/source-branch-hooks/` prefix
+* A script stored in a GitHub repository using the `git:` prefix.
+    * Format: `git:https_repository_clone_url/branch:repo/relative/path/to/script`
+
+### Environment variables in source branch hook
+
+- **`REBASEBOT_SOURCE_REPO`** - Source repository in the format `namespace/repository`.
+
+#### Example
+
+```sh
+rebasebot ... \
+    --source-repo namespace/source-repo \
```

Comment:
Could we use a full blown github url in the example here? To be consistent with the `--source` entry.